src/commonMain/kotlin/org/kbson/BsonMaxKey.kt

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.kbson
+
+/** Represent the maximum key value regardless of the key's type */
+class BsonMaxKey : BsonValue() {
+    override fun getBsonType(): BsonType {
+        return BsonType.MAX_KEY
+    }
+
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (other == null || this::class != other::class) return false
+        return true
+    }
+
+    override fun hashCode(): Int {
+        return this::class.hashCode()
```

Comment:
If only a single instance of this is ever supported, you can turn this class into a singleton using `object BsonMaxKey` instead?

Reply:
Done